pkg/apis/serving/k8s_validation_test.go

```diff
@@ -336,7 +336,50 @@ func TestPodSpecMultiContainerValidation(t *testing.T) {
 			Paths:   []string{"containers[0].ports"},
 			Details: "Only a single port is allowed",
 		}),
-	}}
+	}, {
+		name: "flag enabled: multiple containers with illegal env variable defined for side car",
+		ps: corev1.PodSpec{
+			Containers: []corev1.Container{{
+				Image: "busybox",
+				Ports: []corev1.ContainerPort{{
+					ContainerPort: 8888,
+				}},
+			}, {
+				Image: "helloworld",
+				Env: []corev1.EnvVar{{
+					Name:  "PORT",
+					Value: "Foo",
+				}, {
+					Name:  "K_SERVICE",
+					Value: "Foo",
+				}},
+			}},
+		},
+		cfgOpts: []configOption{withMultiContainer()},
+		want: &apis.FieldError{
+			Message: `"K_SERVICE" is a reserved environment variable`,
+			Paths:   []string{"containers[1].env[1].name"},
+		},
+	},
+		{
```

Comment:
nit:

```suggestion
	}, {
```

Tiny little golang formatting "trick" to tame the indentation a bit better.